ext/storage_sources/azure_store/azure_connection.cpp

```diff
@@ -58,9 +61,18 @@ azure_connection::list_objects(
     return 0;
 }
 
+// Puts an object given a reference to the filename.
 int
 azure_connection::put_object(const std::string &file_name) const
 {
+    std::filesystem::path p(file_name);
+    bool exists = std::filesystem::exists(p);
+    if (!exists) {
+        std::cerr << file_name << ": No such file exists." << std::endl;
+        return -1;
+    }
+    auto blob_client = _azure_client.GetBlockBlobClient(_object_prefix + p.filename().u8string());
```

Comment:
Currently unsure if this is the best way - At the moment it only takes the filename + extension. Doesn't consider what if we have a/test.py and test.py since this would view them as both test.py. A better solution would be to pass in the file path as opposed to the file name however this would need a root / reference point